music_assistant/providers/webdav/provider.py

```diff
@@ -0,0 +1,1040 @@
+                        is_dir=True,
+                    )
+                raise MediaNotFoundError(f"WebDAV resource not found: {file_path}")
+
+            webdav_item = items[0]
+
+            return FileSystemItem(
+                filename=PurePosixPath(file_path).name or webdav_item.name,
+                relative_path=file_path,
+                absolute_path=webdav_url,
+                is_dir=webdav_item.is_dir,
+                checksum=webdav_item.last_modified or "unknown",
+                file_size=webdav_item.size,
+            )
+
+        except MediaNotFoundError:
+            raise
+        except Exception as err:
+            raise MediaNotFoundError(f"Failed to resolve WebDAV path {file_path}: {err}") from err
+
+    def get_absolute_path(self, file_path: str) -> str:
+        """Return WebDAV URL for given file path."""
+        return build_webdav_url(self.base_url, file_path)
+
+    async def browse(self, path: str) -> Sequence[MediaItemType | ItemMapping | BrowseFolder]:
+        """Browse this provider's items."""
+        item_path = path.split("://", 1)[1] if "://" in path else ""
+
+        # For all paths (including subpaths for audiobooks/podcasts), browse actual folders
+        items: list[MediaItemType | ItemMapping | BrowseFolder] = []
+
+        try:
+            filesystem_items = await self._scandir(item_path)
+
+            for item in filesystem_items:
+                if not item.is_dir and ("." not in item.filename or not item.ext):
+                    continue
+
+                if item.is_dir:
+                    items.append(
+                        BrowseFolder(
+                            item_id=item.relative_path,
+                            provider=self.instance_id,
+                            path=f"{self.instance_id}://{item.relative_path}",
+                            name=item.filename,
+                            is_playable=True,
+                        )
+                    )
+                elif item.ext in TRACK_EXTENSIONS:
+                    items.append(
+                        ItemMapping(
+                            media_type=MediaType.TRACK,
+                            item_id=item.relative_path,
+                            provider=self.instance_id,
+                            name=item.filename,
+                        )
+                    )
+                elif item.ext in PLAYLIST_EXTENSIONS:
+                    items.append(
+                        ItemMapping(
+                            media_type=MediaType.PLAYLIST,
+                            item_id=item.relative_path,
+                            provider=self.instance_id,
+                            name=item.filename,
+                        )
+                    )
+        except Exception as err:
+            self.logger.error(f"Failed to browse WebDAV path {item_path}: {err}")
+
+        return items
+
+    async def get_track(self, prov_track_id: str) -> Track:
```

Comment:
this does not have to be overwritten from the base implementation imo

Reply:
After implementing your earlier suggestion which removed the auth_url line then indeed this can be removed